common/postgresql-ng/templates/configmap.yaml

```diff
@@ -0,0 +1,69 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "fullname" . }}-etc
+data:
+  postgresql.conf: |
+    {{- include (print .Template.BasePath "/etc/_postgresql.conf.tpl") . | nindent 4 }}
+
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "fullname" . }}-init-db
+data:
+  init-db.sql: |
+    {{ .Values.initDb }}
+
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "fullname" . }}-maintain-db
+data:
+{{- $users := .Values.users }}
+{{- if .Values.persistence.createBackupUser }}
+{{- $backup = nil }}
+{{- $users = $backup }}
+{{- end }}
+{{- if .Values.createMetricsUser }}
+{{- $metrics = nil }}
+{{- $users = $metrics }}
+{{- end }}
```

Comment:
What is the purpose of this structure? It looks broken to me. For example, `if .Values.createMetricsUser`, then `$metrics` is set to nil and then therefore `$users` is also set to nil, which would make the entire following `range` ineffective.

Reply:
I was searching for the dict function and didn't know how to do it....